mingw-w64-git/PKGBUILD

```diff
@@ -75,8 +77,8 @@ sha256sums=('SKIP'
 if test -z "$WITHOUT_PDBS"
 then
 	pkgname+=("${MINGW_PACKAGE_PREFIX}-${_realname}-pdb")
-	makedepends+=("${MINGW_PACKAGE_PREFIX}-cv2pdb")
-	options+=('!strip')
+	[[ "$MSYSTEM" == "CLANGARM64" ]] || makedepends+=("${MINGW_PACKAGE_PREFIX}-cv2pdb")
+	[[ "$MSYSTEM" == "CLANGARM64" ]] || options+=('!strip')
```

Comment:
This is an interesting one, so we don't need `cv2pdb`.

- Clang generates `.pdb` files for us (see `LDFLAGS ...` below)
- `makepkg` strips the binaries by default

I tried using `LDFLAGS = -Wl,--pdb= -Wl,--strip-all`, but that doesn't actually strip the binaries. When I take out the `--pdb` part, the binaries _are_ stripped, but then obviously the PDB files aren't generated.

What is the main intention here? To simply generate `.pdb` files, write those into the `mingw-w64-x86_64-git-pdb-...` package, and then strip the actual binaries?

I'm considering to always use `options+=('!strip')` here, and then manually invoke `llvm-strip` at the end, since `makepkg` built-in strip command is incredibly slow (probably due to x64 emulation). [Should I do](https://llvm.org/docs/CommandGuide/llvm-strip.html) a `llvm-strip --strip-all` or `llvm-strip --strip-debug`? Would it still be possible to debug with the `.pdb`s when doing `--strip-all`?

Reply:
> ```diff
> +	[[ "$MSYSTEM" == "CLANGARM64" ]] || makedepends+=("${MINGW_PACKAGE_PREFIX}-cv2pdb")
> +	[[ "$MSYSTEM" == "CLANGARM64" ]] || options+=('!strip')
> ```

Would you mind turning this into a proper `if` block, i.e.

```sh
if test CLANGARM64 = "$MSYSTEM"
then
    [...]
else
    makedepends+=("${MINGW_PACKAGE_PREFIX}-cv2pdb")
    options+=('!strip')
fi

> What is the main intention here?

To provide `.pdb` files for all the `.exe` files, so that crashes can be diagnosed more efficiently.

> use `options+=('!strip')` here, and then manually invoke `llvm-strip` at the end

That sounds like the most sensible thing to do, as it aligns the clang code path with the non-clang one.

> [Should I do](https://llvm.org/docs/CommandGuide/llvm-strip.html) a `llvm-strip --strip-all` or `llvm-strip --strip-debug`?

It sounds very much to me as if `--strip-debug` is the equivalent to `options+=('strip')`.